### PR TITLE
Add solution for #2053

### PR DESCRIPTION
**Pull request by viet2307**
Added solution for #2053
- Go solution contributed by @lyxuansang91

### CODE REVIEW

**Comment by viet2307** [comment]
Added solution for #502 IPO

**Review by Hth4nh — COMMENTED** [review]
Good, will be merged soon

File: Easy/2053. Kth Distinct String in an Array/solution.md
Comment (lyxuansang91) [inline Easy/2053. Kth Distinct String in an Array/solution.md:1]: modify file name to match our convention in README.md

File: Hard/502. IPO/solution.md
Comment (lyxuansang91) [inline Hard/502. IPO/solution.md:1]: The same with above

**Pull request merged**